analyzers/src/SonarAnalyzer.CSharp/Rules/TestClassShouldHaveTestMethod.cs

```diff
@@ -65,19 +65,19 @@ public sealed class TestClassShouldHaveTestMethod : SonarDiagnosticAnalyzer
                 SyntaxKind.ClassDeclaration,
                 SyntaxKindEx.RecordClassDeclaration);
 
-        private static bool HasAnyTestMethod(INamedTypeSymbol classSymbol) =>
-            classSymbol.GetMembers().OfType<IMethodSymbol>().Any(m => m.IsTestMethod());
+        private static bool HasAnyTestMethod(INamespaceOrTypeSymbol symbol) =>
```

Comment:
Why this change from `INamedTypeSymbol` to `INamspaceOrTypeSymbol`? It is not used inside, but from a method contract point of view, this method doesn't make sense for namespaces.

Reply:
I am using the most abstract type for flexibility. No strong preference here.